packages/medusa/src/api/routes/admin/price-lists/list-price-list-products.ts

```diff
@@ -0,0 +1,174 @@
+ *   - (query) collection_id {string[]} Collection ids to search for.
+ *   - (query) tags {string[]} Tags to search for.
+ *   - (query) title {string} to search for.
+ *   - (query) description {string} to search for.
+ *   - (query) handle {string} to search for.
+ *   - (query) is_giftcard {string} Search for giftcards using is_giftcard=true.
+ *   - (query) type {string} to search for.
+ *   - (query) order {string} to retrieve products in.
+ *   - (query) deleted_at {DateComparisonOperator} Date comparison for when resulting products was deleted, i.e. less than, greater than etc.
+ *   - (query) created_at {DateComparisonOperator} Date comparison for when resulting products was created, i.e. less than, greater than etc.
+ *   - (query) updated_at {DateComparisonOperator} Date comparison for when resulting products was updated, i.e. less than, greater than etc.
+ *   - (query) offset {string} How many products to skip in the result.
+ *   - (query) limit {string} Limit the number of products returned.
+ *   - (query) expand {string} (Comma separated) Which fields should be expanded in each product of the result.
+ *   - (query) fields {string} (Comma separated) Which fields should be included in each product of the result.
+ * tags:
+ *   - Product
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *             count:
+ *               description: The number of Products.
+ *               type: integer
+ *             offset:
+ *               description: The offset of the Product query.
+ *               type: integer
+ *             limit:
+ *               description: The limit of the Product query.
+ *               type: integer
+ *             products:
+ *               type: array
+ *               items:
+ *                 $ref: "#/components/schemas/product"
+ */
+export default async (req, res) => {
+  const { id } = req.params
+
+  const validatedParams = await validator(
+    AdminGetPriceListsPriceListProductsParams,
+    req.query
+  )
+
+  req.query.price_list_id = [id]
+
+  const filterableFields: FilterableProductProps = omit(req.query, [
+    "limit",
+    "offset",
+    "expand",
+    "fields",
+    "order",
+  ])
+
+  const result = await listAndCount(
+    req.scope,
+    filterableFields,
+    {},
+    {
+      limit: validatedParams.limit ?? 50,
+      offset: validatedParams.offset ?? 0,
+      expand: validatedParams.expand,
+      fields: validatedParams.fields,
+      order: validatedParams.order,
+      allowedFields: allowedAdminProductFields,
+      defaultFields: defaultAdminProductFields,
+      defaultRelations: defaultAdminProductRelations,
+    }
+  )
+
+  res.json(result)
+}
+
+enum ProductStatus {
+  DRAFT = "draft",
+  PROPOSED = "proposed",
+  PUBLISHED = "published",
+  REJECTED = "rejected",
+}
+
+export class AdminGetPriceListsPriceListProductsParams extends AdminGetProductsPaginationParams {
```

Comment:
Similar to another comment, I think we should only have one place for `FilterableProductParams`, and then use TypeScript's utils like `Omit` to get the exact params for the different use-cases. Otherwise, we are just burdening ourselves with maintenance